### PR TITLE
Prevent child from calling exit() on interrupt

### PR DESCRIPTION
**Pull request by dmz-uk**
A proposed fix for the logrotate SIGHUP issue.

### CODE REVIEW

File: src/child.c [hunk 1]
```diff
@@ -233,6 +233,7 @@ static void child_main (struct child_s *ptr)
 
                 ret = select(maxfd + 1, &rfds, NULL, NULL, NULL);
                 if (ret == -1) {
+                        if (errno == EINTR) continue;
```
Comment (obnoxxx) [inline src/child.c]: The change looks reasonable.
But on the cosmetic side, could you put the continue on a line of its own and use braces even around the 1-line block? Thanks!

**Comment by rofl0r** [comment]
@obnoxxx ^

**Comment by obnoxxx** [comment]
This looks good. Can you squash this with the original commit and (force) push? I'll merge that... Merry Christmas. :-)

**Comment by dmz-uk** [comment]
Sorry for the delay, Christmas and New Year! It's been squashed.

**Comment by obnoxxx** [comment]
LGTM, sorry for the delay..

**Pull request merged**